glean-core/ffi/src/from_raw.rs

```diff
@@ -110,6 +110,38 @@ pub fn from_raw_string_array_and_string_array(
     }
 }
 
+/// Create a HashMap<String, i32> from a pair of C string and int arrays.
+///
+/// Returns an error if any of the strings contain invalid UTF-8 characters.
+///
+/// ## Safety
+///
+/// * We check the array pointer for validity (non-null).
+/// * FfiStr checks each individual char pointer for validity (non-null).
+/// * We discard invalid char pointers (null pointer).
+/// * Invalid UTF-8 in any string will return an error from this function.
+pub fn from_raw_string_array_and_int_array(
+    keys: RawStringArray,
+    values: RawIntArray,
+    len: i32,
+) -> glean_core::Result<Option<HashMap<String, i32>>> {
+    unsafe {
```

Comment:
Can we put the `unsafe` around just the minimal parts that we know require `unsafe`?  For example, I think the first clause here is actually safe, etc.

Reply:
I think we touched this topic with @badboy a while ago and we agreed on wrapping the whole function there (I've copy pasted it, basically).